src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorConfiguration.cs

```diff
@@ -11,20 +12,23 @@ public sealed record class RazorConfiguration(
     RazorLanguageVersion LanguageVersion,
     string ConfigurationName,
     ImmutableArray<RazorExtension> Extensions,
+    bool SuppressAddComponentParameter,
```

Comment:
I went back and forth on that. I don't like optional params for internal only code, but yea... having every callsite know exactly what to pass is verbose

Reply:
> I don't like optional params for internal only code

Is there a reason why? I definitely don't share the same principle and optional parameters are definitely more dangerous for public APIs.

For me, the problem isn't just verbosity. The problem is that anyone constructing a `RazorConfiguration` now needs to know what to pass and there's no documentation. Optional params are useful for providing documentation. If you don't like using optional params for some reason, please add a constructor overload that doesn't take `SuppressAddComponentParameter` and sets the value to false.

Reply:
> If you don't like using optional params for some reason, please add a constructor overload that doesn't take SuppressAddComponentParameter and sets the value to false.

We already have optional parameters, so adding another isn't problematic. 

> Is there a reason why?

It's not enough of a reason to vastly change code structure. In general I find that internal code has value in being explicit about expectations, and optional params don't necessarily tell the story of that to me. Questions like https://github.com/dotnet/razor/pull/10763#discussion_r1722491939 would likely not have come up if I didn't explicitly require a parameter. I didn't answer it because I don't know the answer, but it's a worthy discussion that came from changing expectations of what a RazorConfiguration contains. The value is in that callsites are at least somewhat vetted for the new expectation in a code review.

Reply:
Just to be different, I don't like optional parameters for internal code (or external code :P) either, but I am completely okay with overloads that elide some parameters to make things easier to call.

Reply:
Well, since @ryzngard is the person who most recently added an optional parameter to this class (`LanguageServerFlags`) it seems like we probably shouldn't change the precedent now. 😛 Besides, the cost of explicitness is almost certainly not worth it in this case. By using an optional parameter, this PR shrank from 90 files changed to just 16. So, the pragmatic approach is probably the better one here.

If this record were defined with init-only properties and default values for the properties, I don't think we'd be having this conversation. :smile: Although, I think we'd still want a constructor to take the non-default arguments. Perhaps the best mix would be like so:

```C#
public sealed record class RazorConfiguration(RazorLanguageVersion LanguageVersion, string ConfigurationName)
{
    public ImmutableArray<RazorExtension> Extensions { get; init; } = [];
    public LanguageServerFlags? LanguageServerFlags { get; init; } = null;
    public bool UseConsolidatedViews { get; init; } = false;
    public bool SuppressAddComponentParameter { get; init; } = false;
}
```

Of course, this is still effectively "optional parameters", so I'm guessing both of you dislike this too? 😁